capstone-frontend/src/pages/UserProfile.jsx

```diff
@@ -35,6 +37,20 @@ function UserProfile() {
     fetchUserData();
   }, []);
 
+  useEffect(() => {
+    fetch("http://localhost:5000/usersposts")
+      .then((response) => response.json())
+      .then((data) => {
+        let relevantComments = [];
+        for (const item in data) {
+          if (data[item].userId == userid) {
+            relevantComments.push(data[item]);
+          }
```

Comment:
This filtering should happen on the server. You could do this via the cookie, or via URL param (e.g. `http://localhost:5000/userposts?fetch_all=true`)

Reply:
Will look into this.